aten/src/ATen/TensorGeometry.h

```diff
@@ -113,6 +113,27 @@ struct TORCH_API TensorGeometry {
     return r;
   }
 
+  std::vector<c10::SymInt>& mutable_sizes() {
+    return sizes_;
+  }
+  std::vector<c10::SymInt>& mutable_strides() {
+    return strides_;
+  }
+  c10::SymInt& mutable_storage_offset() {
+    return storage_offset_;
+  }
+  void recompute() {
```

Comment:
Should this be called from the constructor as well?

Reply:
Yes and we compute it alongside of the stride for the contiguous case.
I guess not re-use it is ok then!